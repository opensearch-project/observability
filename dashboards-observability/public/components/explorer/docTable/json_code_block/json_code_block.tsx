/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiCodeBlock } from '@elastic/eui';
import { i18n } from '@osd/i18n';
import { IDocType } from '../docViewRow';

export function JsonCodeBlock({ hit }: { hit: IDocType }) {
  const label = i18n.translate('discover.docViews.json.codeEditorAriaLabel', {
    defaultMessage: 'Read only JSON view of an elasticsearch document',
  });
  return (
    <EuiCodeBlock aria-label={label} language="json" isCopyable paddingSize="s">
      {JSON.stringify(hit, null, 2)}
    </EuiCodeBlock>
  );
}
