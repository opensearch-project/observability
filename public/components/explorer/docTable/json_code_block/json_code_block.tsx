/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { EuiCodeBlock } from '@elastic/eui';
import { i18n } from '@osd/i18n';
import { IDocType } from '../docViewRow';

export function JsonCodeBlock({ hit }: { hit: IDocType }) {
  const label = i18n.translate('discover.docViews.json.codeEditorAriaLabel', {
    defaultMessage: 'Read only JSON view of an elasticsearch document',
  });
  return (
    <EuiCodeBlock aria-label={label} language="json" isCopyable paddingSize="s">
      {JSON.stringify(hit, null, 2)}
    </EuiCodeBlock>
  );
}
