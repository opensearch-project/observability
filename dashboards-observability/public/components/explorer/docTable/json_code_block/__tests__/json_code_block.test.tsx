/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { JsonCodeBlock } from '../json_code_block';

describe('Doc viewer JSON block component', () => {
  configure({ adapter: new Adapter() });

  it('Renders JSON block component', async () => {
    
    const hit = {
      'Carrier': 'JetBeats',
      'avg(FlightDelayMin)': '45.957544288332315'
    };

    const wrapper = mount(
      <JsonCodeBlock 
        hit={hit}
      />
    );
    
    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});