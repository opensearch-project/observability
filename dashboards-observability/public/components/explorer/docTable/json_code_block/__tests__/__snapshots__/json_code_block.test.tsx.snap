// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doc viewer JSON block component Renders JSON block component 1`] = `
<JsonCodeBlock
  hit={
    Object {
      "Carrier": "JetBeats",
      "avg(FlightDelayMin)": "45.957544288332315",
    }
  }
>
  <EuiCodeBlock
    aria-label="Read only JSON view of an elasticsearch document"
    isCopyable={true}
    language="json"
    paddingSize="s"
  >
    <EuiCodeBlockImpl
      aria-label="Read only JSON view of an elasticsearch document"
      inline={false}
      isCopyable={true}
      language="json"
      paddingSize="s"
    >
      <Portal
        containerInfo={
          <div>
            {
  "Carrier": "JetBeats",
  "avg(FlightDelayMin)": "45.957544288332315"
}
          </div>
        }
      >
        {
  "Carrier": "JetBeats",
  "avg(FlightDelayMin)": "45.957544288332315"
}
      </Portal>
      <EuiInnerText
        fallback=""
      >
        <div
          className="euiCodeBlock euiCodeBlock--fontSmall euiCodeBlock--paddingSmall euiCodeBlock--hasControls"
          style={Object {}}
        >
          <pre
            className="euiCodeBlock__pre euiCodeBlock__pre--whiteSpacePreWrap"
            style={Object {}}
          >
            <code
              aria-label="Read only JSON view of an elasticsearch document"
              className="euiCodeBlock__code json"
            />
          </pre>
          <div
            className="euiCodeBlock__controls"
          >
            <div
              className="euiCodeBlock__copyButton"
            >
              <EuiI18n
                default="Copy"
                token="euiCodeBlock.copyButton"
              >
                <EuiCopy
                  afterMessage="Copied"
                  textToCopy="{
  \\"Carrier\\": \\"JetBeats\\",
  \\"avg(FlightDelayMin)\\": \\"45.957544288332315\\"
}"
                >
                  <EuiToolTip
                    delay="regular"
                    onMouseOut={[Function]}
                    position="top"
                  >
                    <span
                      className="euiToolTipAnchor"
                      onKeyUp={[Function]}
                      onMouseOut={[Function]}
                      onMouseOver={[Function]}
                    >
                      <EuiButtonIcon
                        aria-label="Copy"
                        color="text"
                        iconType="copy"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        size="s"
                      >
                        <button
                          aria-label="Copy"
                          className="euiButtonIcon euiButtonIcon--text"
                          disabled={false}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          size="s"
                          type="button"
                        >
                          <EuiIcon
                            aria-hidden="true"
                            className="euiButtonIcon__icon"
                            size="m"
                            type="copy"
                          >
                            <EuiIconEmpty
                              aria-hidden={true}
                              className="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
                              focusable="false"
                              role="img"
                              style={null}
                            >
                              <svg
                                aria-hidden={true}
                                className="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
                                focusable="false"
                                height={16}
                                role="img"
                                style={null}
                                viewBox="0 0 16 16"
                                width={16}
                                xmlns="http://www.w3.org/2000/svg"
                              />
                            </EuiIconEmpty>
                          </EuiIcon>
                        </button>
                      </EuiButtonIcon>
                    </span>
                  </EuiToolTip>
                </EuiCopy>
              </EuiI18n>
            </div>
          </div>
        </div>
      </EuiInnerText>
    </EuiCodeBlockImpl>
  </EuiCodeBlock>
</JsonCodeBlock>
`;
